packages/server/src/integration-test/postgres.spec.ts

```diff
@@ -0,0 +1,663 @@
+          },
+        },
+        description: {
+          name: "description",
+          type: FieldType.STRING,
+        },
+        value: {
+          name: "value",
+          type: FieldType.NUMBER,
+        },
+        linkedField: {
+          type: FieldType.LINK,
+          constraints: {
+            type: "array",
+            presence: false,
+          },
+          fieldName: "foreignField",
+          name: "linkedField",
+          relationshipType: RelationshipTypes.ONE_TO_MANY,
+          tableId: auxPostgresTable._id,
+        },
+      },
+      sourceId: postgresDatasource._id,
+    })
+  })
+
+  afterAll(async () => {
+    await config.end()
+  })
+
+  function makeRandomRow() {
+    return {
+      name: generator.name(),
+      description: generator.paragraph(),
+      value: generator.age(),
+    }
+  }
+
+  async function createRow(
+    row: {
+      name: string
+      description: string
+      value: number
+    },
+    tableId?: string,
+    createForeignRow?: boolean
+  ) {
+    if (createForeignRow) {
+      const foreignRow = await config.createRow({
+        tableId: auxPostgresTable._id,
+        title: generator.name(),
+      })
+
+      row = {
+        ...row,
+        [`fk_${auxPostgresTable.name}_foreignField`]: foreignRow.id,
+      }
+    }
+
+    return await config.createRow({
+      tableId: tableId || postgresTable._id,
+      ...row,
+    })
+  }
+
+  async function createDefaultPgTable() {
+    return await config.createTable({
+      name: generator.word({ length: 10 }),
+      type: "external",
+      primary: ["id"],
+      schema: {
+        id: {
+          name: "id",
+          type: FieldType.AUTO,
+          constraints: {
+            presence: true,
+          },
+        },
+      },
+      sourceId: postgresDatasource._id,
+    })
+  }
+
+  async function populateRows(count: number, tableId?: string) {
+    return await Promise.all(
+      Array(count)
+        .fill({})
+        .map(async () => {
+          const rowData = makeRandomRow()
+          return {
+            rowData,
+            row: await createRow(rowData, tableId || postgresTable._id),
+          }
+        })
+    )
+  }
+
+  test("validate table schema", async () => {
+    const res = await makeRequest(
+      "get",
+      `/api/datasources/${postgresDatasource._id}`
+    )
+
+    expect(res.status).toBe(200)
+    expect(res.body).toEqual({
+      config: {
+        ca: false,
+        database: "postgres",
+        host,
+        password: "--secret-value--",
+        port,
+        rejectUnauthorized: false,
+        schema: "public",
+        ssl: false,
+        user: "postgres",
+      },
+      plus: true,
+      source: "POSTGRES",
+      type: "datasource",
+      _id: expect.any(String),
+      _rev: expect.any(String),
+      createdAt: expect.any(String),
+      updatedAt: expect.any(String),
+      entities: expect.any(Object),
+    })
+  })
+
+  describe("create a row", () => {
```

Comment:
Personal preference when writing api `describe` blocks is to use the verb / endpoint combination. I find it's easier to grep the test file for the block I want and sets a easy to extend standard. ie:

```ts
describe("POST /api/:tableId/rows")
describe("DELETE /api/:tableId/rows")
```

This isn't very common across our tests, but it's something new I've found useful.

Reply:
I think it's a good idea for API tests!